Add sensei preview course

rawrrencee opened the pull request:
will update once obj is updated on backend side to return firstNme/lastName/profileImgUrl
addresses #118 

from telegram: _can i request for Reviews of a course to also join the User table to return basic details (firstName, lastName, profileImgUrl) of the user that wrote the review?_

# Changelog:

Add Sensei view Course as Student functionality
Add all reviews modal to show all reviews of the course (to be updated when review object returns the basic user details about who posted the review)
Remove the comments tab from the Create Lesson modal.
Align profile pic and Reviews box.

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

## Screenshots (where relevant):
<img width="985" alt="image" src="https://user-images.githubusercontent.com/43084055/114266557-79a31680-9a29-11eb-8010-3fda41460966.png">
<img width="966" alt="image" src="https://user-images.githubusercontent.com/43084055/114266562-7f006100-9a29-11eb-8113-c1b69e32ae39.png">
<img width="1004" alt="image" src="https://user-images.githubusercontent.com/43084055/114266563-832c7e80-9a29-11eb-8b12-7049eee1036f.png">

rawrrencee committed:
add placeholder Reviews modal to store reviews & paginatedReviews

will update once obj is updated on backend side to return firstNme/lastName/profileImgUrl

rawrrencee committed:
remove comments tab from create/update lesson in course (sensei)

rawrrencee committed:
align items in review comment

All reviews modal section to be updated with actual data when Nat is done with the review portion

rawrrencee committed:
update admin/sensei course banner to be non-scrollable but clickable to view

rawrrencee committed:
Merge branch 'develop' into add-sensei-preview-course

natalieloke commented on src/pages/student/dashboard/courses/view/index.js:
```diff
@@ -120,17 +124,30 @@ const StudentCourseDetails = () => {
+                icon={<FormOutlined />}
+              >
+                Reviews
+              </Button>
```
other students cannot see reviews of the same course? 

also wondering if the FormOutlined icon would indicate the idea of filling in a form but we are just clicking on it to view reviews

rawrrencee replied:
oh yeah! okay then i'll just remove the reviews portion and add it in again later as a separate Reviews card!

natalieloke reviewed (commented):
in terms of UX, when I think in terms of a user (student/admin), i would probably prefer to view all reviews at one shot when I am viewing the course! means like it's not nested in the modal in the button and it's in the same level as the course description etc. 

@rawrrencee  what do you think?

rawrrencee committed:
remove view all reviews modal

TODO: add in ReviewsCard component to display all Reviews for a Course in the main page of the Course

rawrrencee commented:
Hey @natalieloke I removed the Reviews button + AllReviewsModal from the View Course page. Will add in a AllReviews card component in the main page instead, which will have pagination later on when backend introduces the User data to each Review object!

rawrrencee commented on src/pages/student/dashboard/courses/view/index.js:
```diff
@@ -120,17 +122,19 @@ const StudentCourseDetails = () => {
-            >
-              {`${editMode ? 'Edit your' : 'Add a'}  Review`}
-            </Button>
+            {user.userType === USER_TYPE_ENUM.STUDENT && (
+              <Button
+                type="primary"
+                size="large"
+                shape="round"
+                onClick={() => {
+                  setShowReviewModal(true)
+                }}
+                icon={<EditOutlined />}
+              >
+                {`${editMode ? 'Edit your' : 'Add a'}  Review`}
+              </Button>
+            )}
```
Left the userType check here since Admin/Sensei cannot review a course

rawrrencee replied:
@natalieloke

natalieloke replied:
Yep I saw this in the earlier review and this is fine

natalieloke commented:
> Hey @natalieloke I removed the Reviews button + AllReviewsModal from the View Course page. Will add in a AllReviews card component in the main page instead, which will have pagination later on when backend introduces the User data to each Review object!

Thanks, will the AllReviews card component be coming in a separate PR? @rawrrencee

rawrrencee commented:
> > Hey @natalieloke I removed the Reviews button + AllReviewsModal from the View Course page. Will add in a AllReviews card component in the main page instead, which will have pagination later on when backend introduces the User data to each Review object!
> 
> Thanks, will the AllReviews card component be coming in a separate PR? @rawrrencee

Yup, won't be in this one! This one just for making the Sensei be able to Preview Course like how a student will see it, and respond to comments inside. (which was sth raised by prof in the 2nd sys rel presentation)
